app/API/ImgUpload.php

```diff
@@ -0,0 +1,48 @@
+<?
+namespace Gazelle\API;
+
+class ImgUpload extends AbstractAPI {
+    public function run() {
+
+        $response = array();
+
+        // Fetching the userid of the user by api key
+        $LoggedUser = $this->cache->get_value("api_apps_{$_GET['api_key']}")[0];
+        if (!is_array($LoggedUser)) {
+            $json_response["error"] = "Unauthorized Access";
+            return $json_response;
+        }
+        
+        // Checking whether there is an image
+        if (empty($_POST['name']) || empty($_POST['urls'])) {
+            $response["Error"] = "Invalid Request";
+            return $response;
+        }
+
+        $user_id = $LoggedUser['UserID'];
+        $names = $_POST['name'];
```

Comment:
Its just a simple endpoint to rehost images. 
`url` could be any image urls. It could have file name or could be a short one. Thus it accepts the name of the image and its url.

Reply:
file name is related to the OSS storage path,  so the `name` should not pass through to the image provider directly, you can refer to the implementation of `sections/upload/imgupload_handler.php`.

but I recommend do not expose `name` field.  because it's redundant.

Reply:
resolved mate